### PR TITLE
implement clear button functionality

### PR DESCRIPTION
**Pull request by lavong**
project such wow, much awesome. i can haz contribute?

### CODE REVIEW

File: woodstorage/src/main/java/com/jordylangen/woodstorage/FileStorage.java [hunk 1]
```diff
@@ -145,6 +145,18 @@ public void call(LogEntry logEntry) {
         return replaySubject.asObservable();
     }
 
+    @Override
+    public void clear() {
+        try {
+            replaySubject = null;
```
Comment (JordyLangen) [inline woodstorage/src/main/java/com/jordylangen/woodstorage/FileStorage.java]: what happens to whichever objects are subscribed at this point?

File: woodstorage/src/main/java/com/jordylangen/woodstorage/view/WoodStoragePresenter.java [hunk 2]
```diff
@@ -33,12 +33,14 @@ public void teardown() {
...
+        unsubscribe();
         if (itemId == R.id.woodstorage_action_sort) {
             invertSortOrder();
-            unsubscribe();
-            view.clear();
-            subscribe();
+        } else if (itemId == R.id.woodstorage_action_clear) {
+            WoodStorageFactory.getWorker().getStorage().clear();
         }
+        view.clear();
```
Comment (JordyLangen) [inline woodstorage/src/main/java/com/jordylangen/woodstorage/view/WoodStoragePresenter.java]: pressing filter now gives you kinda weird behaviour, but thats fine :)

File: woodstorage/src/main/java/com/jordylangen/woodstorage/view/WoodStoragePresenter.java [hunk 3]
```diff
@@ -33,12 +33,14 @@ public void teardown() {
...
     @Override
     public void onOptionsItemSelected(int itemId) {
+        unsubscribe();
         if (itemId == R.id.woodstorage_action_sort) {
             invertSortOrder();
-            unsubscribe();
-            view.clear();
-            subscribe();
+        } else if (itemId == R.id.woodstorage_action_clear) {
+            WoodStorageFactory.getWorker().getStorage().clear();
```
Comment (JordyLangen) [inline woodstorage/src/main/java/com/jordylangen/woodstorage/view/WoodStoragePresenter.java]: Assuming there is a worker 😲

File: woodstorage/src/test/groovy/com/jordylangen/woodstorage/InMemoryStorageSpec.groovy [hunk 4]
```diff
@@ -18,18 +18,38 @@ class InMemoryStorageSpec extends RxSpecification {
...
         !logs.isEmpty()
         logs.size() == InMemoryStorage.MAX_ITEMS
         logs[0].message == Integer.toString(InMemoryStorage.MAX_ITEMS)
         logs[InMemoryStorage.MAX_ITEMS - 1].message == Integer.toString((InMemoryStorage.MAX_ITEMS * 2) - 1)
     }
+
+    def "should clear logs"() {
+        when:
+        for (def i = 0; i < InMemoryStorage.MAX_ITEMS * 2; i++) {
+            storage.save(new LogEntry("test", Log.DEBUG, Integer.toString(i), null))
```
Comment (JordyLangen) [inline woodstorage/src/test/groovy/com/jordylangen/woodstorage/InMemoryStorageSpec.groovy]: not sure if we really need to add `MAX_ITEMS * 2` for this testcase ?

File: woodstorage/src/test/groovy/com/jordylangen/woodstorage/view/WoodStoragePresenterSpec.groovy [hunk 5]
```diff
@@ -81,4 +81,19 @@ class WoodStoragePresenterSpec extends RxSpecification {
...
+
+        when:
+        presenter.setup(view)
+
+        and:
+        presenter.onOptionsItemSelected(R.id.woodstorage_action_clear)
+
+        then:
+        1 * storage.clear()
+        1 * view.clear()
```
Comment (JordyLangen) [inline woodstorage/src/test/groovy/com/jordylangen/woodstorage/view/WoodStoragePresenterSpec.groovy]: `2 * storage.load() >> Observable.empty()`?

**Comment by lavong** [comment]
good points! fixed.

File: woodstorage/src/main/java/com/jordylangen/woodstorage/view/WoodStoragePresenter.java (hunk 3 above)
Comment (lavong) [inline woodstorage/src/main/java/com/jordylangen/woodstorage/view/WoodStoragePresenter.java]: haha, the word `factory` had me there 😃

File: woodstorage/src/main/java/com/jordylangen/woodstorage/FileStorage.java [hunk 6]
```diff
@@ -145,6 +145,21 @@ public void call(LogEntry logEntry) {
         return replaySubject.asObservable();
     }
 
+    @Override
+    public void clear() {
+        try {
+            if (replaySubject != null) {
+                replaySubject.onCompleted();
```
Comment (JordyLangen) [inline woodstorage/src/main/java/com/jordylangen/woodstorage/FileStorage.java]: :+1:

**Pull request merged**